themes/default/content/learn/pulumi-fundamentals/export-outputs/index.md

```diff
@@ -0,0 +1,256 @@
+we configure values in Pulumi, they are specific to a stack. So, let's set the
+`node_environment` to `production` in the prod stack and add all the other ports
+and configuration variable values back to our environment:
+
+```bash
+pulumi config set frontend_port 3001
+pulumi config set backend_port 3000
+pulumi config set mongo_port 27017
+pulumi config set mongo_host mongodb://mongo:27017
+pulumi config set database cart
+pulumi config set node_environment production
+```
+
+Remember that just because you're on a different stack, the ports are still
+taken by the previous stack. We'll need to destroy the dev stack before the prod
+stack can stand up here as we've hardwired the ports into the Docker image. You
+could avoid this situation by modifying your Dockerfile to pull the port from
+your Pulumi program. For now, though, let's switch back over to the dev stack
+and take it down.
+
+You can list your stacks with `pulumi stack ls` again:
+
+```bash
+pulumi stack ls
+NAME   LAST UPDATE    RESOURCE COUNT  URL
+dev    7 minutes ago  0               https://app.pulumi.com/spara/my-first-app/dev
+prod*  3 minutes ago  9               https://app.pulumi.com/spara/my-first-app/prod
+```
+
+The asterisk (*) indicates that this stack is the current stack for your
+terminal environment. To switch to another stack, you can use the `select`
+command, so let's switch back to dev:
+
+```bash
+pulumi stack select dev
+```
+
+Then, run `pulumi destroy` to take down the current infrastructure in dev:
+
+```bash
+pulumi destroy
+```
+
+Once that command completes, switch back to prod:
+
+```bash
+pulumi stack select prod
+```
+
+Now that you have the basics of creating, selecting, examining stacks, and
+exporting stack outputs, run `pulumi up` again. You should get new images and
+containers, this time running in production mode!
+
+You can also list the current stack's resources with `pulumi stack`:
+
+```bash
+pulumi stack
+Current stack is prod:
+    Owner: spara
+    Last updated: 2 hours ago (2021-07-10 17:44:24.657842983 -0500 CDT)
+    Pulumi version: v3.6.1
+Current stack resources (9):
+    TYPE                                         NAME
+    pulumi:pulumi:Stack                          my-first-app-prod
+    ├─ docker:image:Image                        backend
+    ├─ docker:image:Image                        frontend
+    ├─ docker:index/network:Network              network
+    ├─ docker:index/remoteImage:RemoteImage      mongo
+    ├─ docker:index/container:Container          mongo_container
+    ├─ docker:index/container:Container          frontend_container
+    ├─ docker:index/container:Container          backend_container
+    └─ pulumi:providers:docker                   default_3_0_0
+
+Current stack outputs (1):
+    OUTPUT        VALUE
+    container_id  ac3312e2ce48d6a268f8a05a757e4fdbd4c520d3b5bd7e3de23130563a33e456
+
+More information at: https://app.pulumi.com/spara/my-first-app/prod
+```
+
+## Stack References
```

Comment:
In my experience, Stack References are too complex to introduce at this stage in the learning process. I think we should remove this

Reply:
FWIW I think we had already discussed that stack references, etc, were going to be pulled out bc it's the main focus of the Pulumi in Practice tutorial.

Reply:
Done; it's now in "in practice"